src/ore/src/num.rs

```diff
@@ -125,3 +125,641 @@ impl fmt::Display for NonNegError {
+                self.0.is_zero()
+            }
+        }
+
+        #[cfg(feature = "num-traits")]
+        impl num_traits::identities::One for Overflowing<$t> {
+            #[inline(always)]
+            fn one() -> Self {
+                Self::ONE
+            }
+        }
+
+        #[cfg(feature = "num-traits")]
+        impl num_traits::Num for Overflowing<$t> {
+            type FromStrRadixErr = <$t as num_traits::Num>::FromStrRadixErr;
+
+            #[inline(always)]
+            fn from_str_radix(str: &str, radix: u32) -> Result<Self, Self::FromStrRadixErr> {
+                <$t>::from_str_radix(str, radix).map(Self)
+            }
+        }
+
+        #[cfg(feature = "num-traits")]
+        impl num_traits::sign::Signed for Overflowing<$t> {
+            #[inline(always)]
+            fn abs(&self) -> Self {
+                Self(self.0.abs())
+            }
+            #[inline(always)]
+            fn abs_sub(&self, other: &Self) -> Self {
+                Self(self.0.abs_sub(&other.0))
+            }
+            #[inline(always)]
+            fn signum(&self) -> Self {
+                Self(self.0.signum())
+            }
+            #[inline(always)]
+            fn is_positive(&self) -> bool {
+                self.0.is_positive()
+            }
+            #[inline(always)]
+            fn is_negative(&self) -> bool {
+                self.0.is_negative()
+            }
+        }
+    };
+}
+macro_rules! impl_overflowing_signed {
+    ($t:ty, $u:ty) => {
+        impl Overflowing<$t> {
+            /// Returns the absolute value of the number as an unsigned integer.
+            #[inline(always)]
+            pub fn unsigned_abs(self) -> $u {
+                self.0.unsigned_abs()
+            }
+        }
+    };
+}
+
+// TODO: Support unsigned integers.
+impl_overflowing!(i8, bool i8, bool,);
+impl_overflowing_signed!(i8, u8);
+impl_overflowing!(i16, bool i8 u8 i16, bool i8 u8,);
+impl_overflowing_signed!(i16, u16);
+impl_overflowing!(i32, bool i8 u8 i16 u16 i32, bool i8 u8 i16 u16, );
+impl_overflowing_signed!(i32, u32);
+// N.B. We're including `isize` here because we know it's 64 bits on all platforms we support.
+impl_overflowing!(i64, bool i8 u8 i16 u16 i32 u32 i64, bool i8 u8 i16 u16 i32 u32, isize);
+impl_overflowing_signed!(i64, u64);
+impl_overflowing!(i128, bool i8 u8 i16 u16 i32 u32 i64 u64 i128, bool i8 u8 i16 u16 i32 u32 i64 u64, isize usize);
+impl_overflowing_signed!(i128, u128);
+
+mod overflowing_support {
+    use std::sync::atomic::AtomicUsize;
+
+    use crate::num::OverflowingBehavior;
+
+    /// Ignore overflow.
+    const MODE_IGNORE: usize = 0;
+    /// Soft assert on overflow.
+    const MODE_SOFT_PANIC: usize = 1;
+    /// Panic on overflow.
+    const MODE_PANIC: usize = 2;
+
+    static OVERFLOWING_MODE: AtomicUsize = AtomicUsize::new(MODE_IGNORE);
+
+    /// Handles overflow for [`Overflowing`](super::Overflowing) numbers.
+    #[track_caller]
+    #[inline]
```

Comment:
This was mildly surprising to me - I would have guessed we'd want `#[cold]/#[inline(never)]`, since that's what Rust tags on [similar error handling paths](https://doc.rust-lang.org/src/alloc/vec/mod.rs.html#1977-1983), but I've got no direct experience one way or the other. What's the reasoning here?

Reply:
No, you're right, it was a blip of mind! It should certainly be cold. Fixing, and thanks for spotting!